cmd/tether/attach.go

```diff
@@ -384,13 +421,37 @@ func (t *attachServerSSH) run() error {
 			}
 
 			sessionid := string(bytes)
-			session, ok := t.config.Sessions[sessionid]
-			if !ok {
+
+			s, oks := t.config.Sessions[sessionid]
+			e, oke := t.config.Execs[sessionid]
+			if !oks && !oke {
 				detail := fmt.Sprintf("session %s is invalid", sessionid)
 				attachchan.Reject(ssh.Prohibited, detail)
 				log.Error(detail)
 				continue
 			}
+			// we have sessionid
+			session := s
+			if oke {
+				session = e
+			}
+
+			// session is potentially blocked in launch until we've got the unblock message, so we cannot lock it.
+			// check that session is valid
+			// The detail remains concise as it'll eventually make its way to the user
+			if session.Started != "" && session.Started != "true" {
+				detail := fmt.Sprintf("launch failed with: %s", session.Started)
```

Comment:
Do we have any way of testing this, or the Stopped case below?
IIRC the personality logic will probably block us from exercising these paths in it's current form.